File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> obj;
    string answer = "";
    if (!completion.size())
        return participant[0];

    for (auto part : participant) {
        if (!obj[part]) obj[part] = 0;
        ++obj[part];
    };

    for (auto part : completion) {
        --obj[part];
    };
    
    for (auto iter = obj.begin(); iter != obj.end(); ++iter) {
        if (iter->second == 1) answer = iter->first;
    };

    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const obj = {}
    var answer = '';

    if (!completion || !completion.length) return participant[0]
    
    for (const part of participant) {
        if (!obj[part]) obj[part] = 0
        ++obj[part]
    }

    for (const part of completion) {
        --obj[part]
    }

    for (let [part, num] of Object.entries(obj)) {
        if (num === 1) answer = part
    }

    return answer;
}